xds/googledirectpath/googlec2p.go

```diff
@@ -166,6 +147,33 @@ func (r *c2pResolver) Close() {
 	r.client.Close()
 }
 
+var (
+	defaultNode = &v3corepb.Node{
+		// Not all required fields are set in defaultNote. Metadata will be set
+		// if ipv6 is enabled. Locality will be set to the value from metadata.
+		Id:                   "C2P",
+		UserAgentName:        gRPCUserAgentName,
+		UserAgentVersionType: &v3corepb.Node_UserAgentVersion{UserAgentVersion: grpc.Version},
+		ClientFeatures:       []string{clientFeatureNoOverprovisioning},
+	}
+	ipv6EnabledMetadata = &structpb.Struct{
+		Fields: map[string]*structpb.Value{
+			ipv6CapableMetadataName: structpb.NewBoolValue(true),
+		},
+	}
+)
+
+// cloneNode makes a copy of defaultNode, and populate it's Metadata and
+// Locality fields.
+func cloneNode(zone string, ipv6Capable bool) *v3corepb.Node {
```

Comment:
This isn't quite what I had in mind.  More like:
```go
func newDefaultNode(zone string, ipv6Capable bool) *v3corepb.Node {
  n := &v3corepb.Node{
		// Not all required fields are set in defaultNote. Metadata will be set
		// if ipv6 is enabled. Locality will be set to the value from metadata.
		Id:                   "C2P",
		UserAgentName:        gRPCUserAgentName,
		UserAgentVersionType: &v3corepb.Node_UserAgentVersion{UserAgentVersion: grpc.Version},
		ClientFeatures:       []string{clientFeatureNoOverprovisioning},
		Locality: &v3corepb.Locality{Zone: zone}
  }
  if ipv6Capable {
    n.Metadata = ipv6EnabledMetadata
  }
  return n
}
```

Reply:
Done